api/client/cli.go

```diff
@@ -160,6 +162,13 @@ func NewDockerCli(in io.ReadCloser, out, err io.Writer, clientFlags *cli.ClientF
 		}
 		cli.configFile = configFile
 
+		// Set the CLI's API version
+		if ver := os.Getenv("DOCKER_API_VERSION"); ver != "" {
+			api.Version = version.Version(ver)
+		} else if configFile.APIVersion != "" {
+			api.Version = version.Version(configFile.APIVersion)
```

Comment:
Correct, no validation.  The point of this is to allow the user of the CLI to change this field. If they set it incorrectly then its their problem.  I have a real need for this with something I'm working on, where I need to set the version string to "XXX", and yes that's valid for my scenario/testing.